packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/eventbridge-scheduler/create-schedule.ts

```diff
@@ -0,0 +1,364 @@
+  /**
+   * The retry policy settings
+   */
+  public retryPolicy?: RetryPolicy;
+  /**
+   * Dead letter queue for failed events
+   */
+  public deadLetterQueue?: sqs.IQueue;
+
+  constructor(props: EventBridgeSchedulerTargetProps) {
+    this.validateProps(props);
+    this.arn = props.arn;
+    this.role = props.role;
+    this.input = props.input;
+    this.retryPolicy = props.retryPolicy;
+    this.deadLetterQueue = props.deadLetterQueue;
+  }
+
+  /**
+   * return the target object for the EventBridge Scheduler
+   */
+  public renderTargetObject(): any {
+    return {
+      Arn: this.arn,
+      RoleArn: this.role.roleArn,
+      Input: this.input,
+      RetryPolicy: this.retryPolicy ? {
+        MaximumEventAgeInSeconds: this.retryPolicy.maximumEventAge.toSeconds(),
+        MaximumRetryAttempts: this.retryPolicy.maximumRetryAttempts,
+      } : undefined,
+      DeadLetterConfig: this.deadLetterQueue ? {
+        Arn: this.deadLetterQueue.queueArn,
+      } : undefined,
+    };
+  }
+
+  private validateProps(props: EventBridgeSchedulerTargetProps) {
+    if (props.input !== undefined && !Token.isUnresolved(props.input) && props.input.length < 1) {
+      throw new Error('Input must be at least 1 character long.');
+    }
+
+    if (props.retryPolicy) {
+      if (
+        !Number.isInteger(props.retryPolicy.maximumRetryAttempts) ||
+        props.retryPolicy.maximumRetryAttempts < 0 ||
+        props.retryPolicy.maximumRetryAttempts > 185
+      ) {
+        throw new Error(`MaximumRetryAttempts must be an integer between 0 and 185, got ${props.retryPolicy.maximumRetryAttempts}`);
+      }
+      if (
+        props.retryPolicy.maximumEventAge.toMilliseconds() < Duration.seconds(60).toMilliseconds() ||
+        props.retryPolicy.maximumEventAge.toSeconds() > 86400
+      ) {
+        throw new Error('MaximumEventAgeInSeconds must be between 60 and 86400 seconds');
+      }
+    }
+  }
+}
+
+/**
+ * The information about the retry policy settings
+ */
+export interface RetryPolicy {
+  /**
+   * The maximum number of retry attempts to make before the request fails.
+   */
+  readonly maximumRetryAttempts: number;
+
+  /**
+   * The maximum amount of time to continue to make retry attempts.
+   */
+  readonly maximumEventAge: Duration;
+}
+
+/**
+ * Create a new AWS EventBridge Scheduler schedule
+ *
+ * @see https://docs.aws.amazon.com/scheduler/latest/APIReference/API_CreateSchedule.html
+ */
+export class EventBridgeSchedulerCreateScheduleTask extends sfn.TaskStateBase {
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EventBridgeSchedulerCreateScheduleTaskProps) {
+    super(scope, id, props);
+
+    this.validateProps(props);
+
+    if (props.kmsKey) {
+      props.kmsKey.grantDecrypt(props.target.role);
+    }
```

Comment:
```suggestion
    props.kmsKey?.grantDecrypt(props.target.role);
```
Should be equivalent.